File: app/src/main/java/com/startandroid/utils/FileUtils.java
package com.startandroid.utils;

import android.content.Context;

import org.jetbrains.annotations.NotNull;

import java.io.File;

public class FileUtils {
    public static void deleteOffline(@NotNull Context context) {
        File archive = new File(context.getFilesDir() + "/offline.zip");
        File resourcesDir = new File(context.getFilesDir() + "/resources");
        try {
            if (archive.exists()) {
                archive.delete();
            }
            if (resourcesDir.exists()) {
                deleteDirectory(resourcesDir);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static boolean deleteDirectory(@NotNull File directory) {
        if (directory.exists()) {
            File[] files = directory.listFiles();
            if (null != files) {
                for (int i = 0; i < files.length; i++) {
                    if (files[i].isDirectory()) {
                        deleteDirectory(files[i]);
                    } else {
                        files[i].delete();
                    }
                }
            }
        }
        return (directory.delete());
    }
}

File: app/src/main/java/com/startandroid/utils/FileUtils.kt
package com.startandroid.utils

import android.content.Context
import java.io.File

object FileUtils {
    fun deleteOffline(context: Context) {
        val archive = File(context.filesDir.toString() + "/offline.zip")
        val resourcesDir = File(context.filesDir.toString() + "/resources")
        try {
            if (archive.exists()) {
                archive.delete()
            }
            if (resourcesDir.exists()) {
                deleteDirectory(resourcesDir)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    @JvmStatic
    fun deleteDirectory(directory: File): Boolean {
        if (directory.exists()) {
            val files = directory.listFiles()
            if (null != files) {
                for (i in files.indices) {
                    if (files[i].isDirectory) {
                        deleteDirectory(files[i])
                    } else {
                        files[i].delete()
                    }
                }
            }
        }
        return directory.delete()
    }
}